File: api/routers/zpa.py
"""ZPA API router.

Each endpoint resolves a tenant, builds the ZPA client, and delegates to
the ZPAService layer — the same layer used by the CLI and headless scripts.
"""

from typing import Any, Dict, Optional

from fastapi import APIRouter, Depends, HTTPException, Response
from pydantic import BaseModel

from api.schemas.zpa import (
    ApplicationEnabledPatch,
    CertificateRotateRequest,
    ConnectorEnabledPatch,
    ConnectorGroupCreate,
    ConnectorGroupEnabledPatch,
    ConnectorNamePatch,
    PRAConsoleEnabledPatch,
    PRAPortalCreate,
    PRAPortalEnabledPatch,
    ServiceEdgeEnabledPatch,
)
from api.dependencies import require_auth, AuthUser

router = APIRouter()


def _get_service(tenant_name: str, user: AuthUser):
    from lib.auth import ZscalerAuth
    from lib.zpa_client import ZPAClient
    from services.config_service import decrypt_secret, get_tenant
    from services.zpa_service import ZPAService
    from api.dependencies import check_tenant_access

    tenant = get_tenant(tenant_name)
    if not tenant:
        raise HTTPException(status_code=404, detail=f"Tenant '{tenant_name}' not found")
    if not tenant.zpa_customer_id:
        raise HTTPException(status_code=400, detail=f"Tenant '{tenant_name}' has no ZPA Customer ID")
    check_tenant_access(tenant.id, user)

    auth = ZscalerAuth(
        tenant.zidentity_base_url,
        tenant.client_id,
        decrypt_secret(tenant.client_secret_enc),
        govcloud=bool(tenant.govcloud),
    )
    client = ZPAClient(auth, tenant.zpa_customer_id, tenant.oneapi_base_url)
    return ZPAService(client, tenant_id=tenant.id)


def _get_db_context(tenant_name: str, user: AuthUser):
    """Returns the tenant object for DB-only read endpoints.

    Validates tenant exists and user has access. No ZPAClient needed.
    """
    from services.config_service import get_tenant
    from api.dependencies import check_tenant_access

    tenant = get_tenant(tenant_name)
    if not tenant:
        raise HTTPException(status_code=404, detail=f"Tenant '{tenant_name}' not found")
    check_tenant_access(tenant.id, user)
    return tenant


# ------------------------------------------------------------------
# Certificates
# ------------------------------------------------------------------

@router.get("/{tenant}/certificates")
def list_certificates(tenant: str, user: AuthUser = Depends(require_auth)):
    """List all certificates for a ZPA tenant."""
    return _get_service(tenant, user).list_certificates()


@router.delete("/{tenant}/certificates/{cert_id}")
def delete_certificate(tenant: str, cert_id: str, user: AuthUser = Depends(require_auth)):
    """Delete a certificate by ID."""
    success = _get_service(tenant, user).delete_certificate(cert_id)
    return {"deleted": success}


@router.post("/{tenant}/certificates/rotate")
def rotate_certificate(tenant: str, req: CertificateRotateRequest, user: AuthUser = Depends(require_auth)):
    """Certificate rotation is not supported via the web API. Use the CLI."""
    raise HTTPException(
        status_code=400,
        detail="Certificate rotation is not supported via the web API. Use the CLI (`zs-config`).",
    )


# ------------------------------------------------------------------
# Applications
# NOTE: These endpoints remain live-API calls (not DB-first) because
# the web UI uses them as part of a create/edit form flow that requires
# real-time data. Candidates for DB-first migration in a future branch.
# ------------------------------------------------------------------

@router.get("/{tenant}/applications")
def list_applications(tenant: str, app_type: str = "BROWSER_ACCESS", user: AuthUser = Depends(require_auth)):
    return _get_service(tenant, user).list_applications(app_type)


@router.get("/{tenant}/applications/{app_id}")
def get_application(tenant: str, app_id: str, user: AuthUser = Depends(require_auth)):
    return _get_service(tenant, user).get_application(app_id)


@router.post("/{tenant}/applications", status_code=201)
def create_application(tenant: str, body: Dict[str, Any], user: AuthUser = Depends(require_auth)):
    return _get_service(tenant, user).create_application(**body)


@router.put("/{tenant}/applications/{app_id}")
def update_application(tenant: str, app_id: str, body: Dict[str, Any], user: AuthUser = Depends(require_auth)):
    return _get_service(tenant, user).update_application(app_id, body)


@router.delete("/{tenant}/applications/{app_id}")
def delete_application(tenant: str, app_id: str, user: AuthUser = Depends(require_auth)):
    success = _get_service(tenant, user).delete_application(app_id, app_name=app_id)
    return {"deleted": success}


@router.patch("/{tenant}/applications/{app_id}/enabled")
def patch_application_enabled(
    tenant: str,
    app_id: str,
    body: ApplicationEnabledPatch,
    user: AuthUser = Depends(require_auth),
):
    return _get_service(tenant, user).set_application_enabled(app_id, body.enabled)


# ------------------------------------------------------------------
# Reference data (DB-first — for create/edit form dropdowns)
# ------------------------------------------------------------------

@router.get("/{tenant}/segment-groups")
def list_segment_groups(tenant: str, user: AuthUser = Depends(require_auth)):
    return _get_service(tenant, user).list_segment_groups()


@router.get("/{tenant}/server-groups")
def list_server_groups(tenant: str, user: AuthUser = Depends(require_auth)):
    return _get_service(tenant, user).list_server_groups()


# ------------------------------------------------------------------
# App Connectors (DB-first list + mutations)
# ------------------------------------------------------------------

@router.get("/{tenant}/app-connectors")
def list_app_connectors(tenant: str, user: AuthUser = Depends(require_auth)):
    """List all app connectors (DB-first). Run Import Config to populate the cache."""
    return _get_service(tenant, user).list_connectors_from_db()


@router.get("/{tenant}/connectors")
def list_connectors(tenant: str, q: Optional[str] = None, user: AuthUser = Depends(require_auth)):
    """List all app connectors (DB-first) with optional name search."""
    return _get_service(tenant, user).list_connectors_from_db(q=q)


@router.patch("/{tenant}/connectors/{connector_id}/enabled")
def patch_connector_enabled(
    tenant: str,
    connector_id: str,
    body: ConnectorEnabledPatch,
    user: AuthUser = Depends(require_auth),
):
    """Enable or disable an app connector."""
    return _get_service(tenant, user).set_connector_enabled(connector_id, body.enabled)


@router.patch("/{tenant}/connectors/{connector_id}/name")
def patch_connector_name(
    tenant: str,
    connector_id: str,
    body: ConnectorNamePatch,
    user: AuthUser = Depends(require_auth),
):
    """Rename an app connector."""
    return _get_service(tenant, user).rename_connector(connector_id, body.name)


@router.delete("/{tenant}/connectors/{connector_id}")
def delete_connector(
    tenant: str,
    connector_id: str,
    user: AuthUser = Depends(require_auth),
):
    """Delete an app connector."""
    svc = _get_service(tenant, user)
    # Resolve name from DB for audit log before deletion
    rows = svc.list_connectors_from_db()
    name = next((r.get("name", connector_id) for r in rows if r.get("zpa_id") == connector_id), connector_id)
    svc.delete_connector(connector_id, name)
    return {"deleted": True}


# ------------------------------------------------------------------
# Connector Groups
# ------------------------------------------------------------------

@router.get("/{tenant}/connector-groups")
def list_connector_groups(tenant: str, q: Optional[str] = None, user: AuthUser = Depends(require_auth)):
    """List all connector groups (DB-first) with optional name search."""
    return _get_service(tenant, user).list_connector_groups_from_db(q=q)


@router.post("/{tenant}/connector-groups", status_code=201)
def create_connector_group(
    tenant: str,
    body: ConnectorGroupCreate,
    user: AuthUser = Depends(require_auth),
):
    """Create a new connector group."""
    return _get_service(tenant, user).create_connector_group(body.name, body.description)


@router.patch("/{tenant}/connector-groups/{group_id}/enabled")
def patch_connector_group_enabled(
    tenant: str,
    group_id: str,
    body: ConnectorGroupEnabledPatch,
    user: AuthUser = Depends(require_auth),
):
    """Enable or disable a connector group."""
    return _get_service(tenant, user).set_connector_group_enabled(group_id, body.enabled)


@router.delete("/{tenant}/connector-groups/{group_id}")
def delete_connector_group(
    tenant: str,
    group_id: str,
    user: AuthUser = Depends(require_auth),
):
    """Delete a connector group."""
    svc = _get_service(tenant, user)
    rows = svc.list_connector_groups_from_db()
    name = next((r.get("name", group_id) for r in rows if r.get("zpa_id") == group_id), group_id)
    svc.delete_connector_group(group_id, name)
    return {"deleted": True}


# ------------------------------------------------------------------
# Service Edges (DB-first list + enable/disable)
# ------------------------------------------------------------------

@router.get("/{tenant}/service-edges")
def list_service_edges(tenant: str, q: Optional[str] = None, user: AuthUser = Depends(require_auth)):
    """List all service edges (DB-first) with optional name search."""
    return _get_service(tenant, user).list_service_edges_from_db(q=q)


@router.patch("/{tenant}/service-edges/{edge_id}/enabled")
def patch_service_edge_enabled(
    tenant: str,
    edge_id: str,
    body: ServiceEdgeEnabledPatch,
    user: AuthUser = Depends(require_auth),
):
    """Enable or disable a service edge."""
    return _get_service(tenant, user).set_service_edge_enabled(edge_id, body.enabled)


# ------------------------------------------------------------------
# PRA Portals (DB-first list + mutations)
# ------------------------------------------------------------------

@router.get("/{tenant}/pra-portals")
def list_pra_portals(tenant: str, q: Optional[str] = None, user: AuthUser = Depends(require_auth)):
    """List all PRA portals (DB-first) with optional name search."""
    return _get_service(tenant, user).list_pra_portals_from_db(q=q)


@router.post("/{tenant}/pra-portals", status_code=201)
def create_pra_portal(
    tenant: str,
    body: PRAPortalCreate,
    user: AuthUser = Depends(require_auth),
):
    """Create a new PRA portal."""
    return _get_service(tenant, user).create_pra_portal(
        name=body.name,
        domain=body.domain,
        certificate_id=body.certificate_id,
        enabled=body.enabled,
        description=body.description,
        user_notification_enabled=body.user_notification_enabled,
        user_notification=body.user_notification,
    )


@router.patch("/{tenant}/pra-portals/{portal_id}/enabled")
def patch_pra_portal_enabled(
    tenant: str,
    portal_id: str,
    body: PRAPortalEnabledPatch,
    user: AuthUser = Depends(require_auth),
):
    """Enable or disable a PRA portal."""
    return _get_service(tenant, user).set_pra_portal_enabled(portal_id, body.enabled)


@router.delete("/{tenant}/pra-portals/{portal_id}")
def delete_pra_portal(
    tenant: str,
    portal_id: str,
    user: AuthUser = Depends(require_auth),
):
    """Delete a PRA portal."""
    svc = _get_service(tenant, user)
    rows = svc.list_pra_portals_from_db()
    name = next((r.get("name", portal_id) for r in rows if r.get("zpa_id") == portal_id), portal_id)
    svc.delete_pra_portal(portal_id, name)
    return {"deleted": True}


# ------------------------------------------------------------------
# PRA Consoles (DB-first list + mutations)
# ------------------------------------------------------------------

@router.get("/{tenant}/pra-consoles")
def list_pra_consoles(tenant: str, q: Optional[str] = None, user: AuthUser = Depends(require_auth)):
    """List all PRA consoles (DB-first) with optional name search."""
    return _get_service(tenant, user).list_pra_consoles_from_db(q=q)


@router.patch("/{tenant}/pra-consoles/{console_id}/enabled")
def patch_pra_console_enabled(
    tenant: str,
    console_id: str,
    body: PRAConsoleEnabledPatch,
    user: AuthUser = Depends(require_auth),
):
    """Enable or disable a PRA console."""
    return _get_service(tenant, user).set_pra_console_enabled(console_id, body.enabled)


@router.delete("/{tenant}/pra-consoles/{console_id}")
def delete_pra_console(
    tenant: str,
    console_id: str,
    user: AuthUser = Depends(require_auth),
):
    """Delete a PRA console."""
    svc = _get_service(tenant, user)
    rows = svc.list_pra_consoles_from_db()
    name = next((r.get("name", console_id) for r in rows if r.get("zpa_id") == console_id), console_id)
    svc.delete_pra_console(console_id, name)
    return {"deleted": True}


# ------------------------------------------------------------------
# Access Policy (DB-first, read-only)
# ------------------------------------------------------------------

@router.get("/{tenant}/access-policy/rules")
def list_access_policy_rules(
    tenant: str,
    q: Optional[str] = None,
    user: AuthUser = Depends(require_auth),
):
    """List all access policy rules (DB-first), sorted by rule_order."""
    return _get_service(tenant, user).list_access_policy_rules_from_db(q=q)


@router.get("/{tenant}/access-policy/rules/export.csv")
def export_access_policy_csv(
    tenant: str,
    user: AuthUser = Depends(require_auth),
):
    """Export access policy rules as CSV (DB-first)."""
    csv_data = _get_service(tenant, user).export_access_policy_csv()
    return Response(
        content=csv_data,
        media_type="text/csv",
        headers={"Content-Disposition": f'attachment; filename="access_policy_{tenant}.csv"'},
    )


# ------------------------------------------------------------------
# Identity (DB-first, read-only)
# ------------------------------------------------------------------

@router.get("/{tenant}/saml-attributes")
def list_saml_attributes(tenant: str, q: Optional[str] = None, user: AuthUser = Depends(require_auth)):
    """List SAML attributes (DB-first)."""
    return _get_service(tenant, user).list_saml_attributes_from_db(q=q)


@router.get("/{tenant}/scim-attributes")
def list_scim_attributes(tenant: str, q: Optional[str] = None, user: AuthUser = Depends(require_auth)):
    """List SCIM user attributes (DB-first)."""
    return _get_service(tenant, user).list_scim_attributes_from_db(q=q)


@router.get("/{tenant}/scim-groups")
def list_scim_groups(tenant: str, q: Optional[str] = None, user: AuthUser = Depends(require_auth)):
    """List SCIM groups (DB-first)."""
    return _get_service(tenant, user).list_scim_groups_from_db(q=q)

File: web/src/api/zpa.ts
import { apiFetch, getAuthHeaders } from "./client";

export interface ZpaCertificate {
  id: string;
  name: string;
  description?: string;
  issued_to?: string;
  issued_by?: string;
  valid_to_in_epoch_sec?: number;
  status?: string;
}

export interface ZpaApplication {
  id: string;
  name: string;
  enabled: boolean;
  applicationType?: string;
  domainNames?: string[];
}

export interface ZpaPraPortal {
  zpa_id: string;
  id?: string;
  name: string;
  domain?: string;
  certificate_id?: string;
  certificate_name?: string;
  enabled?: boolean;
  [key: string]: unknown;
}

export interface ZpaSegmentGroup {
  id: string;
  name: string;
  [key: string]: unknown;
}

export interface ZpaServerGroup {
  id: string;
  name: string;
  [key: string]: unknown;
}

export interface ZpaAppConnector {
  zpa_id: string;
  id?: string;
  name: string;
  enabled?: boolean;
  [key: string]: unknown;
}

export interface ZpaServiceEdge {
  zpa_id: string;
  id?: string;
  name: string;
  enabled?: boolean;
  [key: string]: unknown;
}

export interface ZpaConnectorGroup {
  zpa_id: string;
  id?: string;
  name: string;
  enabled?: boolean;
  [key: string]: unknown;
}

export interface ZpaPraConsole {
  zpa_id: string;
  id?: string;
  name: string;
  enabled?: boolean;
  [key: string]: unknown;
}

export interface ZpaAccessPolicyRule {
  zpa_id: string;
  id?: string;
  name: string;
  action?: string;
  rule_order?: number;
  [key: string]: unknown;
}

export interface ZpaSamlAttribute {
  zpa_id: string;
  id?: string;
  name: string;
  [key: string]: unknown;
}

export interface ZpaScimAttribute {
  zpa_id: string;
  id?: string;
  name: string;
  [key: string]: unknown;
}

export interface ZpaScimGroup {
  zpa_id: string;
  id?: string;
  name: string;
  [key: string]: unknown;
}

const base = (tenant: string) => `/api/v1/zpa/${encodeURIComponent(tenant)}`;

export const fetchCertificates = (tenant: string): Promise<ZpaCertificate[]> =>
  apiFetch<ZpaCertificate[]>(`${base(tenant)}/certificates`);

export const fetchApplications = (tenant: string, appType = "BROWSER_ACCESS"): Promise<ZpaApplication[]> =>
  apiFetch<ZpaApplication[]>(`${base(tenant)}/applications?app_type=${encodeURIComponent(appType)}`);

export const fetchPraPortals = (tenant: string): Promise<ZpaPraPortal[]> =>
  apiFetch<ZpaPraPortal[]>(`${base(tenant)}/pra-portals`);

// ── Application mutations ──────────────────────────────────────────────────────

export const createApplication = (tenant: string, body: unknown): Promise<ZpaApplication> =>
  apiFetch<ZpaApplication>(`${base(tenant)}/applications`, {
    method: "POST",
    body: JSON.stringify(body),
  });

export const updateApplication = (
  tenant: string,
  appId: string,
  body: unknown
): Promise<ZpaApplication> =>
  apiFetch<ZpaApplication>(`${base(tenant)}/applications/${encodeURIComponent(appId)}`, {
    method: "PUT",
    body: JSON.stringify(body),
  });

export const deleteApplication = (
  tenant: string,
  appId: string
): Promise<{ deleted: boolean }> =>
  apiFetch<{ deleted: boolean }>(`${base(tenant)}/applications/${encodeURIComponent(appId)}`, {
    method: "DELETE",
  });

export const patchApplicationEnabled = (
  tenant: string,
  appId: string,
  enabled: boolean
): Promise<ZpaApplication> =>
  apiFetch<ZpaApplication>(`${base(tenant)}/applications/${encodeURIComponent(appId)}/enabled`, {
    method: "PATCH",
    body: JSON.stringify({ enabled }),
  });

// ── Segment / Server groups ────────────────────────────────────────────────────

export const listSegmentGroups = (tenant: string): Promise<ZpaSegmentGroup[]> =>
  apiFetch<ZpaSegmentGroup[]>(`${base(tenant)}/segment-groups`);

export const listServerGroups = (tenant: string): Promise<ZpaServerGroup[]> =>
  apiFetch<ZpaServerGroup[]>(`${base(tenant)}/server-groups`);

// ── App Connectors ─────────────────────────────────────────────────────────────

export const listAppConnectors = (tenant: string): Promise<ZpaAppConnector[]> =>
  apiFetch<ZpaAppConnector[]>(`${base(tenant)}/app-connectors`);

export const listConnectors = (tenant: string, q?: string): Promise<ZpaAppConnector[]> =>
  apiFetch<ZpaAppConnector[]>(
    `${base(tenant)}/connectors${q ? `?q=${encodeURIComponent(q)}` : ""}`
  );

export const patchConnectorEnabled = (
  tenant: string,
  id: string,
  enabled: boolean
): Promise<ZpaAppConnector> =>
  apiFetch<ZpaAppConnector>(`${base(tenant)}/connectors/${encodeURIComponent(id)}/enabled`, {
    method: "PATCH",
    body: JSON.stringify({ enabled }),
  });

export const patchConnectorName = (
  tenant: string,
  id: string,
  name: string
): Promise<ZpaAppConnector> =>
  apiFetch<ZpaAppConnector>(`${base(tenant)}/connectors/${encodeURIComponent(id)}/name`, {
    method: "PATCH",
    body: JSON.stringify({ name }),
  });

export const deleteConnector = (tenant: string, id: string): Promise<{ deleted: boolean }> =>
  apiFetch<{ deleted: boolean }>(`${base(tenant)}/connectors/${encodeURIComponent(id)}`, {
    method: "DELETE",
  });

// ── Connector Groups ───────────────────────────────────────────────────────────

export const listConnectorGroups = (tenant: string, q?: string): Promise<ZpaConnectorGroup[]> =>
  apiFetch<ZpaConnectorGroup[]>(
    `${base(tenant)}/connector-groups${q ? `?q=${encodeURIComponent(q)}` : ""}`
  );

export const createConnectorGroup = (
  tenant: string,
  name: string,
  description?: string
): Promise<ZpaConnectorGroup> =>
  apiFetch<ZpaConnectorGroup>(`${base(tenant)}/connector-groups`, {
    method: "POST",
    body: JSON.stringify({ name, description }),
  });

export const patchConnectorGroupEnabled = (
  tenant: string,
  id: string,
  enabled: boolean
): Promise<ZpaConnectorGroup> =>
  apiFetch<ZpaConnectorGroup>(
    `${base(tenant)}/connector-groups/${encodeURIComponent(id)}/enabled`,
    {
      method: "PATCH",
      body: JSON.stringify({ enabled }),
    }
  );

export const deleteConnectorGroup = (tenant: string, id: string): Promise<{ deleted: boolean }> =>
  apiFetch<{ deleted: boolean }>(`${base(tenant)}/connector-groups/${encodeURIComponent(id)}`, {
    method: "DELETE",
  });

// ── Service Edges ──────────────────────────────────────────────────────────────

export const listServiceEdges = (tenant: string): Promise<ZpaServiceEdge[]> =>
  apiFetch<ZpaServiceEdge[]>(`${base(tenant)}/service-edges`);

export const patchServiceEdgeEnabled = (
  tenant: string,
  id: string,
  enabled: boolean
): Promise<ZpaServiceEdge> =>
  apiFetch<ZpaServiceEdge>(`${base(tenant)}/service-edges/${encodeURIComponent(id)}/enabled`, {
    method: "PATCH",
    body: JSON.stringify({ enabled }),
  });

// ── PRA Portals ────────────────────────────────────────────────────────────────

export const patchPraPortalEnabled = (
  tenant: string,
  id: string,
  enabled: boolean
): Promise<ZpaPraPortal> =>
  apiFetch<ZpaPraPortal>(`${base(tenant)}/pra-portals/${encodeURIComponent(id)}/enabled`, {
    method: "PATCH",
    body: JSON.stringify({ enabled }),
  });

export const deletePraPortal = (tenant: string, id: string): Promise<{ deleted: boolean }> =>
  apiFetch<{ deleted: boolean }>(`${base(tenant)}/pra-portals/${encodeURIComponent(id)}`, {
    method: "DELETE",
  });

// ── PRA Consoles ───────────────────────────────────────────────────────────────

export const listPraConsoles = (tenant: string, q?: string): Promise<ZpaPraConsole[]> =>
  apiFetch<ZpaPraConsole[]>(
    `${base(tenant)}/pra-consoles${q ? `?q=${encodeURIComponent(q)}` : ""}`
  );

export const patchPraConsoleEnabled = (
  tenant: string,
  id: string,
  enabled: boolean
): Promise<ZpaPraConsole> =>
  apiFetch<ZpaPraConsole>(`${base(tenant)}/pra-consoles/${encodeURIComponent(id)}/enabled`, {
    method: "PATCH",
    body: JSON.stringify({ enabled }),
  });

export const deletePraConsole = (tenant: string, id: string): Promise<{ deleted: boolean }> =>
  apiFetch<{ deleted: boolean }>(`${base(tenant)}/pra-consoles/${encodeURIComponent(id)}`, {
    method: "DELETE",
  });

// ── Access Policy ──────────────────────────────────────────────────────────────

export const listAccessPolicyRules = (
  tenant: string,
  q?: string
): Promise<ZpaAccessPolicyRule[]> =>
  apiFetch<ZpaAccessPolicyRule[]>(
    `${base(tenant)}/access-policy/rules${q ? `?q=${encodeURIComponent(q)}` : ""}`
  );

export async function exportAccessPolicyCsv(tenant: string, tenantName: string): Promise<void> {
  const res = await fetch(
    `/api/v1/zpa/${encodeURIComponent(tenant)}/access-policy/rules/export.csv`,
    {
      headers: getAuthHeaders(),
      credentials: "include",
    }
  );
  if (!res.ok) throw new Error("Export failed");
  const blob = await res.blob();
  const url = URL.createObjectURL(blob);
  const a = document.createElement("a");
  a.href = url;
  a.download = `access_policy_${tenantName}.csv`;
  a.click();
  URL.revokeObjectURL(url);
}

// ── Identity ───────────────────────────────────────────────────────────────────

export const listSamlAttributes = (tenant: string, q?: string): Promise<ZpaSamlAttribute[]> =>
  apiFetch<ZpaSamlAttribute[]>(
    `${base(tenant)}/saml-attributes${q ? `?q=${encodeURIComponent(q)}` : ""}`
  );

export const listScimAttributes = (tenant: string, q?: string): Promise<ZpaScimAttribute[]> =>
  apiFetch<ZpaScimAttribute[]>(
    `${base(tenant)}/scim-attributes${q ? `?q=${encodeURIComponent(q)}` : ""}`
  );

export const listScimGroups = (tenant: string, q?: string): Promise<ZpaScimGroup[]> =>
  apiFetch<ZpaScimGroup[]>(
    `${base(tenant)}/scim-groups${q ? `?q=${encodeURIComponent(q)}` : ""}`
  );
